app/controllers/concerns/api/v2/parameter_deprecation.rb

```diff
@@ -0,0 +1,61 @@
+# This module adds a pre-action filter that will handle deprecated properties.
```

Comment:
What about simplifying the module, would it be enough?

Most of the complication comes from traversing the `params` tree (`params[:a][:b][:c]`). I can "outsource" it as lambdas in the declaration. Would it be better?

``` ruby
deprecate_param [:create, :update], ->(params) { params[:override_value][:use_puppet_default] }, ->(params, value) { params[:override_value][:skip_foreman] = value }
```